src/pexprs-eval.js

```diff
@@ -285,9 +280,11 @@ pexprs.Apply.prototype.eval = function(state) {
 
   var memoKey = app.toMemoKey();
   var memoRec = posInfo.memo[memoKey];
+  var isTopLevelApplication =
+      !caller && this !== state.applySpaces && this !== pexprs.end;
```

Comment:
Right you are, not sure why I wrote that. Will fix in a follow-up commit tomorrow.